test/GPv2AllowListAuthenticator.test.ts

```diff
@@ -3,69 +3,110 @@ import { Contract } from "ethers";
 import { ethers, waffle } from "hardhat";
 
 describe("GPv2AllowListAuthentication", () => {
-  const [deployer, owner, nonOwner, solver] = waffle.provider.getWallets();
+  const [
+    deployer,
+    owner,
+    manager,
+    newManager,
+    nobody,
+    solver,
+  ] = waffle.provider.getWallets();
   let authenticator: Contract;
 
   beforeEach(async () => {
     const GPv2AllowListAuthentication = await ethers.getContractFactory(
-      "GPv2AllowListAuthentication",
+      "GPv2AllowListAuthenticationTestInterface",
       deployer,
     );
 
-    authenticator = await GPv2AllowListAuthentication.deploy();
-    await authenticator.initializeManager(owner.address);
+    authenticator = await GPv2AllowListAuthentication.deploy(owner.address);
```

Comment:
I was a bit confused here, then I realized that this is not creating a `GPv2AllowListAuthentication` but a `GPv2AllowListAuthenticationTestInterface`, which has a nonempty constructor. I'd add a comment explaining this difference.